src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -279,23 +275,30 @@ object RuleUtils {
             baseOutput,
             _,
             _) =>
-        val curFiles = location.allFiles
-          .map(f => FileInfo(f, index.fileIdTracker.addFile(f)))
-
         val (filesDeleted, filesAppended) =
-          if (HyperspaceConf.hybridScanDeleteEnabled(spark) && index.hasLineageColumn) {
-            val (exist, nonExist) = curFiles.partition(index.sourceFileInfoSet.contains)
-            val filesAppended = nonExist.map(f => new Path(f.name))
-            if (exist.length < index.sourceFileInfoSet.size) {
-              (index.sourceFileInfoSet -- exist, filesAppended)
+          if (!HyperspaceConf.hybridScanEnabled(spark) && index.hasSourceUpdate) {
+            // This index is refreshed as quick mode and validated with the latest signature.
```

Comment:
nit: "... was refreshed in quick mode..."

Reply:
Thanks!